src/atoms/second_order_cone/rational_norm.jl

```diff
@@ -0,0 +1,118 @@
+#############################################################################
+# rational_norm.jl
+#
+# Handles the k-norms for k > 1 where k is rational. Reduces the
+# k-norm constraint to at most 2d ceil(log2(n + m)) + O(d) second
+# order cone constraints. Here d is the dimension of the problem and k
+# is converted to a rational with value k = n / m. The procedure for
+# this is based on the paper "Second-order cone programming" by
+# F. Alizadeh and D. Goldfarb, Mathematical Programming, Series B,
+# 95:3-51, 2001.
+#############################################################################
+
+export rational_norm
+
+### k-norm for rational k
+
+type RationalNormAtom <: AbstractExpr
+  head::Symbol
+  id_hash::Uint64
+  children::(AbstractExpr,)
+  size::(Int, Int)
+  k::Rational
```

Comment:
``Rational`` is an abstract type, so any code that interacts with this field will be very slow. You may want ``Rational{Int64}`` or ``Rational{BigInt}`` instead.